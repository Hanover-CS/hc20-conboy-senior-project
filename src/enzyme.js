/* Author: Patrick Conboy
 * Date Created: 1/1/2019
 * Description: This file initializes Enzyme for testing purposes.
 */

import Enzyme, { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
export { shallow, mount, render };
export default Enzyme;