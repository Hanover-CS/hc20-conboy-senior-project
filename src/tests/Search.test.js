import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { shallow } from '../enzyme'

import Search from '../Search'

describe('Search Tests', () => {
  test('search page renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Search />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test('search page renders correctly', () => {
    const wrapper = shallow(<Search props={""}/>)

    expect(wrapper.find('.props')).toBeDefined()
    expect(wrapper.find('.animeList')).toBeDefined()
  })

  test('snapshot of search page hasn\'t changed', () => {
    const tree = renderer.create(<Search />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})