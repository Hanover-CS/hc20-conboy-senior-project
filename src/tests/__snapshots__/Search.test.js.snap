// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Tests snapshot of search page hasn't changed 1`] = `
<div
  className="searchList"
>
  <section
    className="section"
  >
    <div>
      <div>
        <input
          className="input"
          onChange={[Function]}
          placeholder="Search..."
          type="text"
        />
        <ul />
      </div>
    </div>
  </section>
</div>
`;
